Plugins/GitUIPluginInterfaces/RepositoryHosts/GitProtocol.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Text.RegularExpressions;
+
+namespace GitUIPluginInterfaces.RepositoryHosts
+{
+    public static class GitProtocol
```

Comment:
🤔 
1. The problem with `static` classes is that they impede/complicate testability of implementations that depend on these static classes. 
However the implementation is relatively benign and can remain as is.
2. The public surface of the API is too broad, `IsGitUrl` nor `IsUrlUsingSsh` have any consumers (I've checked #6328 and #6515).
I understand they are added for completeness however we need to be careful adding functionality for sake of adding it.

Reply:
I thought about including these too when looking at the follow up PRs. The current usage is really 'if http else ssh' which is not too bad. I had comments about the completeness but also said I was OK with it (I want to raise such concerns for other reviewers).

Reply:
These aren't blockers, more raising for the general awareness